### ISSUE TITLE
Proposal: compose a roadmap for 2.0

### ISSUE THREAD

**Issue by Kureev** [wix/react-native-navigation#1120]
Dear community,

I'd like to propose to compose a roadmap for 2.0 and make it possible to delegate some stories to the community. For instance, I'd like to help bridging some functionality to the new version. I know, it may sounds like a lot of work, but I strongly believe that it will pays off.

**Comment by Ehesp** [comment]
Hi, there's currently a roadmap for outstanding todos: https://github.com/wix/react-native-navigation/milestone/1

@DanielZlotin is the best person to speak with regarding progress.

**Comment by DanielZlotin** [comment]
Thank you! greatly appreciated!
One of the challenges of working on a project of this magnitude is how do we work together with the community and utilize any help we can get.. in all fairness it is something that we need (and want) to learn!
This is actually what drove us to do a rewrite in the first place. The only way I know to ensure constant quality and be flexible enough so that we can accept PRs from the community without breaking everything (and be able to sleep at night) is complete test coverage. So the rewrite is done with TDD, and no piece of code is written without a failing test (e2e or unit, both JS and native).
Basically, v2.0.0 is for the community.

In terms of current progress, most of the work in the JS side is done, so most of what's left is the actual native handling of everything.

My current short term goal is to publish a minimum viable product of 2.0.0-alpha which has the basic core functionality without any animations/styles, with a deadline of may 7th. Then we can think how we can incorporate your help on this branch and get it up to speed with v1.

**Comment by Kureev** [comment]
Great! Thanks for your feedback.

In the meantime I'll go thought the issues to get an overview of what needs to be done for 2.0. Would be great to triage this roadmap together with interested persons to align everyone with the plan. Then we can try to mark issues as "Good first task" or "Community responsibility" to separate critical issues from ones that community can implement by their own.

Feel free to ping me here or by email any time you need.

**Comment by DanielZlotin** [comment]
I'm trying to think of ways we can interact productively.. we have issues, discord.. maybe we need slack? I don't know

**Comment by guyca** [comment]
We're using [discord](https://discord.gg/DhkZjq2), mainly because reactiflux is there, so we thought that's the go to place for react/react-native devs. Should we switch to slack?

**Comment by Kureev** [comment]
I'm using Slack, but I don't want to push you to switch Discord because of me :)

So I'll install Discord then

**Comment by amay** [comment]
hi @DanielZlotin, any further update on the v2 timeline? Looks like the last activity on this issue was about a year ago. Looking at the milestone, it looks like there is a lot of solid progress: https://github.com/wix/react-native-navigation/milestone/1. I'm in the process of switching off of react-navigation, and while I have been able to get everything working in v1, so far, I have also had to do a fair number of known v1 workarounds to get there. It sounds like v2 should help clean things up nicely, hence my curiosity on timeline. Thanks for the great library!

**Comment by guyca** [comment]
Hey @amay 
We're actually about to publish v2 to beta which means the api is pretty much final. There are still a few missing features, mostly on Android. You can track progress [here](https://github.com/wix/react-native-navigation/projects/3?fullscreen=true).
We've started migrating the Wix app to v2 last week. If your timeline allows for it, I would suggest using v2 or at least migrate to it soon
